comparisons/validateJR.sh

```diff
@@ -49,14 +29,14 @@ grep root ${inList} | grep -v "#" | while read -r dsN fNP procN comm; do
     if [ -f "${baseA}/${fN}" ]; then
 	extN=all_${diffN}_${dsN}
 	mkdir -p ${extN}
-	cd ${cWD}/${extN}
-	cp ${VALIDATE_C_SCRIPT} ./
-	echo "Will run on ${fN} in ${cWD}/${extN}"
-	echo "Now in `pwd`"
-	g++ -shared -o validate.so validate.C `root-config --cflags ` -fPIC
-	echo -e "gSystem->Load(\"libFWCoreFWLite.so\");\n AutoLibraryLoader::enable();\n FWLiteEnabler::enable();\n 
-    .x validate.C+(\"${extN}\", \"${baseA}/${fN}\", \"${baseB}/${fN}\", \"${procN}\");\n .qqqqqq" | root -l -b >& ${extN}.log &
-	waitForProcesses
+	pushd ${extN}
+	  cp ${VALIDATE_C_SCRIPT} ./
+	  echo "Will run on ${fN} in ${extN}"
+	  g++ -shared -o validate.so validate.C `root-config --cflags ` -fPIC
+	  echo -e "gSystem->Load(\"libFWCoreFWLite.so\");\n AutoLibraryLoader::enable();\n FWLiteEnabler::enable();\n
+                  .x validate.C+(\"${extN}\", \"${baseA}/${fN}\", \"${baseB}/${fN}\", \"${procN}\");\n .qqqqqq" | root -l -b >& ${extN}.log &
+        popd
+        while [ $(jobs -p | wc -l) -ge ${nProc} ] ; do sleep 5 ; done
```

Comment:
this would count everything else started outside of of this script and also sub-processes started by ```root```. The ones that are just waiting for some active child to finish will make it look like the node is more busy than it really is.
I recall having to decouple the count of ```root``` vs ```root.exe```.

Reply:
No @slava77 , it only counts for the processes it has directly started [a]. So any process started from its parent or from its child are not going to change the number of jobs. I had tested this with simple bach scripts and it works as expacted.

[a]
```
11638          \_ /bin/bash /data/cmsbld/muz/results/JR-comparison/validateJR.sh /data/cmsbld/muz/data/PR-1e2753 /data/cmsbld/muz/data/CMSSW_11_1_X_2020-02-10-2300 OldVSNew matrix_RE.txt
14948              \_ /bin/bash /data/cmsbld/muz/results/JR-comparison/validateJR.sh /data/cmsbld/muz/data/PR-1e2753 /data/cmsbld/muz/data/CMSSW_11_1_X_2020-02-10-2300 OldVSNew matrix_RE.txt
14980                  \_ root -l -b
14995                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
15054                  \_ root -l -b
15065                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
15270                  \_ root -l -b
15279                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
15281                  \_ root -l -b
15292                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16153                  \_ root -l -b
16165                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16240                  \_ root -l -b
16247                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16378                  \_ root -l -b
16386                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16678                  \_ root -l -b
16696                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16779                  \_ root -l -b
16784                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16850                  \_ root -l -b
16854                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16955                  \_ root -l -b
16959                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
16967                  \_ root -l -b
16972                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
17164                  \_ root -l -b
17171                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
17286                  |       \_ sh -c cd "/data/cmsbld/muz/results/JR-comparison/all_OldVSNew_ZEEFS13pHLTwf135p4" ; g++ -fPIC -c -O3 -DNDEBUG -D__ROOFIT_NOBANNER -std=c++17 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe -W -Woverloaded-virtual -fsigned-char -pthread  -I$ROOTSYS/include -I"/data/cmsbld/muz/CMSSW_11_1_PY3_X_2020-02-10-2300/src" -I"/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_1_PY3_X_2020-02-10-2300/src" -
17287                  |           \_ g++ -fPIC -c -O3 -DNDEBUG -D__ROOFIT_NOBANNER -std=c++17 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe -W -Woverloaded-virtual -fsigned-char -pthread -I/cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/include -I/17245                  \_ root -l -b
17252                  |   \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/lcg/root/6.18.04-blocog/bin/root.exe -splash -l -b
17282                  \_ g++ -shared -o validate.so validate.C -pthread -std=c++17 -m64 -I/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_1_PY3_X_2020-02-10-2300/external/slc7_amd64_gcc820/bin/../../../../../../../slc7_amd64_gcc820/lcg/root/6.18.04-blocog/include -fPIC
17283                      \_ /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../libexec/gcc/x86_64-unknown-linux-gnu/8.3.1/cc1plus -quiet -I /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_1_PY3_X_2020-02-10-2300/external/slc7_amd64_gcc820/bin/../../../../../../../slc7_amd64_gcc820/lcg/root/6.18.04-blocog/include -iprefix /cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/external/gcc/8.2.0-pa
```